cranelift/assembler/src/mem.rs

```diff
@@ -164,6 +166,35 @@ impl GprMem {
     }
 }
 
+#[derive(Arbitrary, Clone, Debug)]
+#[allow(clippy::module_name_repetitions)]
+pub enum XmmMem {
+    Xmm(Xmm),
+    Mem(Amode),
+}
```

Comment:
I think all of the rest of this makes sense: it will look a bit different on `main` but we do want to distinguish between `Gpr` and `Xmm` like you do here.